packages/medusa-plugin-sendgrid/src/services/sendgrid.js

```diff
@@ -184,40 +184,15 @@ class SendGridService extends NotificationService {
   }
 
   getTemplateId(event) {
-    switch (event) {
-      case "order.return_requested":
-        return this.options_.order_return_requested_template
-      case "swap.shipment_created":
-        return this.options_.swap_shipment_created_template
-      case "claim.shipment_created":
-        return this.options_.claim_shipment_created_template
-      case "order.items_returned":
-        return this.options_.order_items_returned_template
-      case "swap.received":
-        return this.options_.swap_received_template
-      case "swap.created":
-        return this.options_.swap_created_template
-      case "gift_card.created":
-        return this.options_.gift_card_created_template
-      case "order.gift_card_created":
-        return this.options_.gift_card_created_template
-      case "order.placed":
-        return this.options_.order_placed_template
-      case "order.shipment_created":
-        return this.options_.order_shipped_template
-      case "order.canceled":
-        return this.options_.order_canceled_template
-      case "user.password_reset":
-        return this.options_.user_password_reset_template
-      case "customer.password_reset":
-        return this.options_.customer_password_reset_template
-      case "restock-notification.restocked":
-        return this.options_.medusa_restock_template
-      case "order.refund_created":
-        return this.options_.order_refund_created_template
-      default:
-        return null
-    }
+    const templates = Object.keys(this.options_ ?? {})
+    const normalizedEvent = event.toLowerCase().replace(".", "_")
+    const key = templates.find((template) => {
+      return (
+        normalizedEvent === template ||
+        `${normalizedEvent}_template` === template
+      )
+    })
+    return key ? this.options_[key] : key
```

Comment:
suggestion(nit): `this.options_[key] ?? key`

Reply:
agreed